### PR TITLE
feat(web): support added for index prop in all the components

### PR DESCRIPTION
**Pull request by anik-ghosh-au7**
**Before submitting a pull request,** please make sure the following is done:

- [x] Describe the proposed changes and how it'll improve the library experience.
  - added support for index prop in all the components for ReactiveSearch Web
  - this prop is used to explicitly specify an index to query against for a particular component.
  
- [x] Please make sure that there are no linting errors in the code.

- [x] Add a demo video/gif/screenshot to explain how did you test the fix.
  - https://www.loom.com/share/5818e220848a48b7bbca76bd2c3644fd
  
- [ ] If it is a global change, try to add any side effects that it could have.

- [x] Create a PR to add/update the docs (if needed) at [here](https://github.com/appbaseio/Docs).
  - PR for docs - https://github.com/appbaseio/Docs/pull/197
  
- [x] Create a PR to add/update the storybook (if needed) at [here](https://github.com/appbaseio/playground).
  - PR for Playground - https://github.com/appbaseio/playground/pull/67
  
- PR for reactivecore - https://github.com/appbaseio/reactivecore/pull/99

**Learn more about contributing:** [Contributing guides](https://github.com/appbaseio/reactivesearch/blob/next/.github/CONTRIBUTING.md)

### CODE REVIEW

File: packages/web/src/components/basic/NumberBox.js [hunk 1]
```diff
@@ -235,6 +247,7 @@ NumberBox.defaultProps = {
 	queryFormat: 'gte',
 	style: {},
 	URLParams: false,
+	index: undefined,
```
Comment (bietkul) [inline packages/web/src/components/basic/NumberBox.js]: @anik-ghosh-au7 You can remove it at all places.

File: packages/web/src/components/basic/NumberBox.js [hunk 2]
```diff
@@ -244,6 +257,7 @@ const mapStateToProps = (state, props) => ({
 	selectedValue: state.selectedValues[props.componentId]
 		? state.selectedValues[props.componentId].value
 		: null,
+	config: state.config,
```
Comment (bietkul) [inline packages/web/src/components/basic/NumberBox.js]: @anik-ghosh-au7 Just map the property(`enableAppbase`) which you're using.

**Review by bietkul — CHANGES_REQUESTED** [review]

**Comment by bietkul** [comment]
Fix conflicts too

**Comment by anik-ghosh-au7** [comment]
@bietkul I have made the changes also updated the link for the `reactivecore` PR in the main comment

File: packages/web/src/components/basic/NumberBox.js [hunk 3]
```diff
@@ -209,6 +218,7 @@ NumberBox.propTypes = {
 	selectedValue: types.selectedValue,
 	setQueryOptions: types.funcRequired,
 	setCustomQuery: types.funcRequired,
+	enableAppbase: types.bool,
```
Comment (bietkul) [inline packages/web/src/components/basic/NumberBox.js:221]: @anik-ghosh-au7 Use `VueTypes` instead of `types` for boolean props. There are some issues that would set the default value to `true`. Make that change at all place.

**Review by bietkul — CHANGES_REQUESTED** [review]
@anik-ghosh-au7 Resolve conflicts

File: packages/web/src/components/basic/NumberBox.js (hunk 3 above)
Comment (anik-ghosh-au7) [reply packages/web/src/components/basic/NumberBox.js:221]: @bietkul made the changes

File: packages/web/package.json [hunk 4]
```diff
@@ -47,7 +47,8 @@
 		"react-redux": "^6.0.1",
 		"rheostat": "^2.1.1",
 		"url-search-params-polyfill": "^7.0.0",
-		"xdate": "^0.8.2"
+		"xdate": "^0.8.2",
+		"vue-types": "^1.3.3"
```
Comment (bietkul) [inline packages/web/package.json]: @anik-ghosh-au7 What is the reason for this change?

**Review by bietkul — CHANGES_REQUESTED** [review]
@anik-ghosh-au7 Why are you using `vue-types` in a react project?

File: packages/web/package.json (hunk 4 above)
Comment (anik-ghosh-au7) [reply packages/web/package.json]: @bietkul because you mentioned it in the above comment - https://github.com/appbaseio/reactivesearch/pull/1696#discussion_r657367262

**Review by bietkul — APPROVED** [review]
